ERCS/erc-7776.md

```diff
@@ -0,0 +1,554 @@
+        - name: totalOtherIncomeDetail1
+          type: uint256
+        - name: totalOtherExpensesDetail1
+          type: uint256
+        - name: totalOtherIncomeDetailN
+          type: uint256
+        - name: totalOtherExpensesDetailN
+          type: uint256
+
+```
+
+#### netIncome
+
+Returns net income for the period (Operating Income + Other Income/Expenses - Taxes - Depreciation).
+
+```yaml
+- name: netIncome
+  type: function
+  visibility: public
+  stateMutability: view
+  inputs:
+    - name: startTimestamp
+      type: uint256
+    - name: endTimestamp
+      type: uint256
+  outputs:
+    - name: NetIncomeStruct
+      type: struct
+      fields:
+        - name: totalNetIncome
+          type: uint256
+        - name: optionalAdditionalNetIncomeDetail1
+          type: uint256
+        - name: optionalAdditionalNetIncomeDetailN
+          type: uint256
+
+```
+
+#### earningsPerShare
+
+Returns Earnings Per Share Token (EPS) for the period.
+
+```yaml
+- name: earningsPerShare
+  type: function
+  visibility: public
+  stateMutability: view
+  inputs:
+    - name: startTimestamp
+      type: uint256
+    - name: endTimestamp
+      type: uint256
+  outputs:
+    - name: EPSstruct
+      type: struct
+      fields:
+        - name: basicEPS
+          type: uint256
+        - name: dilutedEPS
+          type: uint256
+        - name: EPSDetail1
+          type: uint256
+        - name: EPSDetailN
+          type: uint256
+
+```
+
+#### fullFinancialReport
+
+Returns a comprehensive struct that includes all the prior financial details of the Protocol combined: Revenue, COGS, Operating Expenses, Operating Income, EBITDA, Other Incomes and Expenses, Net income, and EPS into a unified `Struct`.
+
+```yaml
+- name: fullFinancialReport
+  type: function
+  stateMutability: view
+  visibility: public
+  inputs:
+    - name: startTimestamp
+      type: uint256
+    - name: endTimestamp
+      type: uint256
+  outputs:
+    - name: FullFinancialsStruct
+      type: struct
+      fields:
+        - name: RevenueStruct
+          type: struct
+        - name: COGSStruct
+          type: struct
+        - name: OperatingExpensesStruct
+          type: struct
+        - name: OperatingIncomeStruct
+          type: struct
+        - name: EBITDAstruct
+          type: struct
+        - name: OtherIncomeExpensesStruct
+          type: struct
+        - name: NetIncomeStruct
+          type: struct
+        - name: EPSstruct
+          type: struct
+
+```
+
+## Rationale
+
+The current state of token investment analysis is opaque, with most information presented in an abstract and non-quantitative form. This standard A.P.I. ensures a consistent and reliable way for investors to evaluate blockchain projects based on real financial data published directly on-chain, not just speculative promises. This will establish greater
+trust in the cryptocurrency markets and align token analysis with the standards of traditional equity markets.
```

Comment:
This paragraph probably belongs in Motivation because it's justifying the proposal as a whole. The Rationale section should explain choices made within this document, while the Motivation should convince the reader that the document itself is necessary (and a better choice than any competing proposals.) My favourite analogy is:

> Motivation: We need to build a shed because...
> Rationale: We decided to paint the shed red because...

Reply:
Good point. Understood. Changed.